File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/injection/modules/DatabaseModule.java
package com.crpr.androidcinema.injection.modules;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import io.realm.RealmConfiguration;

/**
 * Created by DEATH_STAR on 25/05/2016.
 */
@Module
public class DatabaseModule {

    @Provides
    RealmConfiguration provideRealmConfiguration(Context context){
        return new RealmConfiguration.Builder(context).build();
    }

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/injection/modules/DatabaseModule.kt
package com.crpr.androidcinema.injection.modules

import android.content.Context

import dagger.Module
import dagger.Provides
import io.realm.RealmConfiguration

/**
 * Created by DEATH_STAR on 25/05/2016.
 */
@Module
class DatabaseModule {

    @Provides
    internal fun provideRealmConfiguration(context: Context): RealmConfiguration {
        return RealmConfiguration.Builder(context).build()
    }

}
